middleware/proxy_1_11.go

```diff
@@ -3,21 +3,44 @@
 package middleware
 
 import (
+	"context"
 	"fmt"
 	"net/http"
 	"net/http/httputil"
+	"strings"
 
 	"github.com/labstack/echo/v4"
 )
 
+// StatusCodeContextCanceled is HTTP status code for when client closed connection
+// regrettably, there is no standard error code for "client closed connection", but
+// for historical reasons we can use a code that a lot of people are already using;
+// using 5xx is problematic for users;
+const StatusCodeContextCanceled = 499
+
 func proxyHTTP(tgt *ProxyTarget, c echo.Context, config ProxyConfig) http.Handler {
 	proxy := httputil.NewSingleHostReverseProxy(tgt.URL)
 	proxy.ErrorHandler = func(resp http.ResponseWriter, req *http.Request, err error) {
 		desc := tgt.URL.String()
 		if tgt.Name != "" {
 			desc = fmt.Sprintf("%s(%s)", tgt.Name, tgt.URL.String())
 		}
-		c.Set("_error", echo.NewHTTPError(http.StatusBadGateway, fmt.Sprintf("remote %s unreachable, could not forward: %v", desc, err)))
+		// if the client canceled the request (usually this means they closed
```

Comment:
Text suggestion:
```
// If the client canceled the request (usually by closing the connection), we can report a 
// client error (4xx) instead of a server error (5xx) to correctly identify the situation. 
// The Go standard library (at of late 2020) wraps the exported, standard 
// context.Canceled error with unexported garbage value requiring a substring check, see
// https://github.com/golang/go/blob/6965b01ea248cabb70c3749fd218b36089a21efb/src/net/net.go#L416-L430
```